uiServices/test/createConfigFile_test.go

```diff
@@ -0,0 +1,202 @@
+
+	response, err := makeRequest(a.client, method, endpoint, body.Content)
+	if err != nil {
+		return err
+	}
+	a.resp = response
+	return nil
+}
+
+func makeRequest(client *http.Client, method, endpoint, body string) (*http.Response, error) {
+
+	var reqBody io.Reader
+	if body != "" {
+		reqBody = strings.NewReader(body)
+	}
+
+	req, err := http.NewRequest(method, "http://localhost:9191"+endpoint, reqBody)
+	req.Header.Set("configPathDir", fmt.Sprintf("%s/src/github.com/xtracdev/automated-perf-test/uiServices/test/",os.Getenv("GOPATH")))
+
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func FeatureContext(s *godog.Suite) {
+	api := &apiFeature{}
+
+	s.BeforeScenario(func(interface{}) {
+
+		api.resetResponse()
+
+	})
+
+	s.Step(`^I send "(GET|POST|PUT|DELETE)" request to "([^"]*)"$`, api.iSendrequestTo)
+	s.Step(`^the response code should be (\d+)$`, api.theResponseCodeShouldBe)
+	s.Step(`^the header configsDirPath is "([^"]*)"$`, theHeaderConfigsDirPathIs)
+	s.Step(`^the response should match json:$`, api.theResponseShouldMatchJSON)
+	s.Step(`^the response body should be empty$`, api.theResponseBodyShouldBeEmpty)
+	s.Step(`^the config file was created at location defined by configsPathDir with parameters:$`, api.theConfigFileWasCreatedAtLocationDefinedByConfigsPathDirWithParameters)
+	s.Step(`^the automated performance ui server is available$`, theAutomatedPerformanceUiServerIsAvailable)
+	s.Step(`^I send "([^"]*)" request to "([^"]*)" with a body:$`, api.iSendRequestToWithABody)
+}
+
+func theAutomatedPerformanceUiServerIsAvailable() error {
+	go http.ListenAndServe(":9191", services.GetRouter())
+	return nil
+}
+
+func IsValidXml(config string) bool{
+	file, err := os.Open(os.Getenv("GOPATH") + "/src/github.com/xtracdev/automated-perf-test/uiServices/test/GodogConfig.xml")
+	if err != nil {
+		fmt.Println(err)
+		return false
+	}
+
+	defer file.Close()
+	logrus.Println("Successfully Opened XML file")
+
+	var actualConfig perfTestUtils.Config
+	var expectedConfig perfTestUtils.Config
+
+	byteValue, err := ioutil.ReadAll(file)
+	xml.Unmarshal(byteValue, &actualConfig)
+	if err != nil{
+		logrus.Println("Cannot Unmarshall into XML")
+		return false
+	}
+
+	bytes, err:= []byte(config),err
+	json.Unmarshal(bytes, &expectedConfig)
+	if err != nil{
+		logrus.Println("Cannot Unmarshall into JSON")
+		return false
+	}
+
+	if
+	!reflect.DeepEqual(&expectedConfig,&actualConfig) ||
+	actualConfig.APIName != expectedConfig.APIName ||
+	actualConfig.AllowablePeakMemoryVariance != expectedConfig.AllowablePeakMemoryVariance ||
+	actualConfig.TargetHost != expectedConfig.TargetHost ||
+	actualConfig.TargetPort != expectedConfig.TargetPort ||
+	actualConfig.MemoryEndpoint != expectedConfig.MemoryEndpoint ||
+	actualConfig.AllowablePeakMemoryVariance != expectedConfig.AllowablePeakMemoryVariance ||
```

Comment:
duplication

Note: run `go vet ./...` on the project and it will tell you if there are any issues.